server/socketserver.go

```diff
@@ -57,18 +56,23 @@ func (s *socketServer) execute(handler Handler, jsonBytes []byte) (reply []byte)
 		return b.Bytes()
 	}
 
-	reply, handlingError := handleRequest(s.repo, handler, jsonBytes, sourceSocketServer)
-	if handlingError != nil {
-		Log.Error("socketServer.execute failed", zap.Error(handlingError))
+	var buf bytes.Buffer
+	buf.Grow(len(jsonBytes))
+	if err := handleRequest(s.repo, handler, bytes.NewReader(jsonBytes), &buf, sourceSocketServer); err != nil {
+		Log.Error("socketServer.execute failed", zap.Error(err))
 	}
-	return reply
+	return buf.Bytes()
 }
 
-func (s *socketServer) writeResponse(conn net.Conn, reply []byte) {
-	headerBytes := []byte(strconv.Itoa(len(reply)))
+func (s *socketServer) writeResponse(w io.Writer, reply []byte) {
+	headerBytes := make([]byte, 0, len(reply)+12)
+	headerBytes = strconv.AppendInt(headerBytes, int64(len(reply)), 10)
 	reply = append(headerBytes, reply...)
-	Log.Debug("replying", zap.String("reply", string(reply)))
-	n, writeError := conn.Write(reply)
+	if Log.Core().Enabled(zap.DebugLevel) {
+		// only log when debug level is really enabled as this costs lots of performance.
```

Comment:
Benchmark or source for this claim?

I would expect zap to run the same check internally.

Reply:
You're allocating memory when casting a byte slice into a string, so this is ZAP independent.

Reply:
Also you should rethink which entries are log worthy and which not and which should change from Debug to Info level, etc.